### PR TITLE
Fix: Issue #81

### PR DESCRIPTION
**Pull request by zhao-stanley**
- Implemented the grid system similar to the Clubs page
- Sorts upcoming contests by date (current --> future) and past contests by most recent
- Added tags at the top left of contests that have passed and contests that are occurring on the day it is viewed on the site
- Upcoming contests are placed at the front while past contests are placed in the back
- Kept the newsletter banner size as the full screen for devices with a width less than 640px and animated an arrow to indicate there is more content below. 
- Devices that are wider than that will have a smaller banner height so that the user can see some contests without scrolling. 

**Images below for further clarification:**

image
image
image
image

### CODE REVIEW

File: src/components/contests/ContestCard.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import Image from "next/dist/client/image"
```
Comment (maggie-j-liu) [inline src/components/contests/ContestCard.tsx:1]:
```suggestion
import Image from "next/image"
```

File: src/components/contests/ContestCard.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          {isPast ? <div className="absolute inset-x-0 top-0 ml-2 mt-2 bg-gray-200 w-min px-2 py-1 rounded-2xl text-black">
+            <p>Past</p>
+          </div> : null}
+          
+          {isToday ? <div className="absolute inset-x-0 top-0 ml-2 mt-2 bg-green-400 w-min px-2 py-1 rounded-2xl text-black">
+            <p>Today</p>
+          </div> : null}
+
+          <div className="absolute inset-x-0 bottom-0 bg-gradient-to-t from-black px-6 pb-6 md:px-4 md:pb-4 lg:px-3 lg:pb-3 pt-12">
+            <p className="mt-2 block md:text-xl text-2xl font-bold text-white truncate pointer-events-none">
```
Comment (maggie-j-liu) [inline src/components/contests/ContestCard.tsx:27]:
```suggestion
            <p className="mt-2 block md:text-xl text-2xl font-bold text-white pointer-events-none">
```
i think it looks a bit better if we don't truncate it?

Comment (zhao-stanley) [reply src/components/contests/ContestCard.tsx:27]: Yeah I was thinking about that. I think we don't truncate it, and if a contest's name is long enough to overflow on mobile, then we can just modify the text size for mobile screens.

Comment (zhao-stanley) [reply src/components/contests/ContestCard.tsx:27]: Even the longer contest names look fine on mobile, so I'll remove the truncate.

**Review by maggie-j-liu — APPROVED** [review]
looks good; thank you!

**Pull request merged**